internal/querynodev2/segments/segment_loader.go

```diff
@@ -704,6 +704,10 @@ func (loader *segmentLoader) LoadDeltaLogs(ctx context.Context, segment *LocalSe
 	var blobs []*storage.Blob
 	for _, deltaLog := range deltaLogs {
 		for _, bLog := range deltaLog.GetBinlogs() {
+			// the segment has applied the delta logs, skip it
+			if bLog.GetTimestampTo() <= segment.LastDeltaTimestamp() {
```

Comment:
handle `GetTimestampTo() == 0`  case, legacy delta log might have this field missing